virt-v2v/cold/constants.go

```diff
@@ -0,0 +1,182 @@
+        exit 1
+    fi
+    echo "Unconnected Ethernet connection: $unconnected_connection"
+}
+
+# Function to copy IPv4 fields
+copy_ipv4_fields() {
+    echo "Copying IPv4 settings from $unconnected_connection to $connection..."
+
+    # Initialize the nmcli command with the connection name
+    nmcli_cmd="nmcli connection modify \"$connection\""
+
+    # Get the ipv4.method
+    ipv4_method=$(nmcli -g ipv4.method connection show "$unconnected_connection")
+    if [ "$ipv4_method" != "--" ]; then
+        nmcli_cmd+=" ipv4.method \"$ipv4_method\""
+    fi
+
+    # Get the ipv4.addresses
+    ipv4_address=$(nmcli -g ipv4.addresses connection show "$unconnected_connection")
+    if [ "$ipv4_address" != "--" ]; then
+        nmcli_cmd+=" ipv4.addresses \"$ipv4_address\""
+    fi
+
+    # Get the ipv4.gateway
+    ipv4_gateway=$(nmcli -g ipv4.gateway connection show "$unconnected_connection")
+    if [ "$ipv4_gateway" != "--" ]; then
+        nmcli_cmd+=" ipv4.gateway \"$ipv4_gateway\""
+    fi
+
+    # Get the ipv4.dns
+    ipv4_dns=$(nmcli -g ipv4.dns connection show "$unconnected_connection")
+    if [ "$ipv4_dns" != "--" ]; then
+        nmcli_cmd+=" ipv4.dns \"$ipv4_dns\""
+    fi
+
+    # Run the built nmcli command if it has modifications
+    if [ "$nmcli_cmd" != "nmcli connection modify \"$connection\"" ]; then
+        eval "$nmcli_cmd"
+    fi
+
+    echo "IPv4 settings copied successfully."
+}
+
+# Function to copy IPv6 fields
+copy_ipv6_fields() {
+    echo "Copying IPv6 settings from $unconnected_connection to $connection..."
+
+    # Initialize the nmcli command with the connection name
+    nmcli_cmd="nmcli connection modify \"$connection\""
+
+    # Get the ipv6.method
+    ipv6_method=$(nmcli -g ipv6.method connection show "$unconnected_connection")
+    if [ "$ipv6_method" != "--" ]; then
+        nmcli_cmd+=" ipv6.method \"$ipv6_method\""
+    fi
+
+    # Get the ipv6.addresses
+    ipv6_address=$(nmcli -g ipv6.addresses connection show "$unconnected_connection")
+    if [ "$ipv6_address" != "--" ]; then
+        nmcli_cmd+=" ipv6.addresses \"$ipv6_address\""
+    fi
+
+    # Get the ipv6.gateway
+    ipv6_gateway=$(nmcli -g ipv6.gateway connection show "$unconnected_connection")
+    if [ "$ipv6_gateway" != "--" ]; then
+        nmcli_cmd+=" ipv6.gateway \"$ipv6_gateway\""
+    fi
+
+    # Get the ipv6.dns
+    ipv6_dns=$(nmcli -g ipv6.dns connection show "$unconnected_connection")
+    if [ "$ipv6_dns" != "--" ]; then
+        nmcli_cmd+=" ipv6.dns \"$ipv6_dns\""
+    fi
+
+    # Run the built nmcli command if it has modifications
+    if [ "$nmcli_cmd" != "nmcli connection modify \"$connection\"" ]; then
+        eval "$nmcli_cmd"
+    fi
+
+    echo "IPv6 settings copied successfully."
+}
+
+# Main script execution
+main() {
+    check_tools
+    identify_device
+    identify_connection
+    find_unconnected_connection
+    copy_ipv4_fields
+    copy_ipv6_fields
+
+    # Restart connection
```

Comment:
This should not be needed, it is also an option to use `nmcli connection reload`, but it also should not be needed